test/e2e/allowed_source_ranges_test.go

```diff
@@ -0,0 +1,396 @@
+	}
+
+	if err := kclient.Get(context.TODO(), name, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	}
+	err := wait.PollImmediate(10*time.Second, 2*time.Minute, func() (bool, error) {
+		cidrs := []string{}
+		if ic.Status.EndpointPublishingStrategy != nil {
+			lb := ic.Status.EndpointPublishingStrategy.LoadBalancer
+			if lb != nil && len(lb.AllowedSourceRanges) > 0 {
+				for _, cidr := range lb.AllowedSourceRanges {
+					cidrs = append(cidrs, string(cidr))
+				}
+			}
+		}
+
+		annotationRanges := strings.Split(lbService.Annotations[ingresscontroller.LoadBalancerSourceRangesAnnotation], ",")
+		t.Logf("Annotation: %v, AllowedSourceRanges: %v", annotationRanges, cidrs)
+		if reflect.DeepEqual(annotationRanges, cidrs) {
+			return true, nil
+		}
+
+		if err := kclient.Get(context.TODO(), name, ic); err != nil {
+			return false, fmt.Errorf("failed to get ingresscontroller: %w", err)
+		}
+
+		return false, nil
+	})
+	if err != nil {
+		t.Fatalf("expected the annotation to be reflected in status.allowedSourceRanges: %v", err)
+	}
+
+	// Set .Spec.LoadBalancerSourceRanges and see if it will be reflected in AllowedSourceRanges in IngressController's status
+	if err := kclient.Get(context.TODO(), controller.LoadBalancerServiceName(ic), lbService); err != nil {
+		t.Fatalf("failed to get service: %v", err)
+	}
+	lbService.Spec.LoadBalancerSourceRanges = []string{"0.0.0.0/0"}
+	if err := kclient.Update(context.TODO(), lbService); err != nil {
+		t.Fatalf("failed to update service: %v", err)
+	}
+
+	if err = kclient.Get(context.TODO(), name, ic); err != nil {
+		t.Fatalf("failed to get ingresscontroller %s: %v", name, err)
+	}
+	err = wait.PollImmediate(10*time.Second, 2*time.Minute, func() (bool, error) {
+		cidrs := []string{}
+		if ic.Status.EndpointPublishingStrategy != nil {
+			lb := ic.Status.EndpointPublishingStrategy.LoadBalancer
+			if lb != nil && len(lb.AllowedSourceRanges) > 0 {
+				for _, cidr := range lb.AllowedSourceRanges {
+					cidrs = append(cidrs, string(cidr))
+				}
+			}
+		}
+
+		t.Logf("LoadBalancerSourceRanges: %v, AllowedSourceRanges: %v", lbService.Spec.LoadBalancerSourceRanges, cidrs)
+
+		if reflect.DeepEqual(lbService.Spec.LoadBalancerSourceRanges, cidrs) {
+			return true, nil
+		}
+
+		if err := kclient.Get(context.TODO(), name, ic); err != nil {
+			return false, fmt.Errorf("failed to get ingresscontroller: %w", err)
+		}
+
+		return false, nil
+	})
+	if err != nil {
+		t.Fatalf("expected the LoadBalancerSourceRanges to be reflected in status.allowedSourceRanges: %v", err)
+	}
+}
+
+// TestSourceRangesAnnotation creates an ingresscontroller with the
+// "LoadBalancerService" endpoint publishing strategy type, adds
+// service.beta.kubernetes.io/load-balancer-source-ranges annotation
+// to the service, and verifies that progressing and evaluation conditions
+// detected statuses of the ingresscontroller are set to true.
+func TestSourceRangesAnnotation(t *testing.T) {
```

Comment:
Would it make sense to combine these tests to reduce the number of LBs we provision?

Reply:
Yes, it'd make sense. I merged the last two tests.